ui/page/wallet_page.go

```diff
@@ -410,6 +425,8 @@ func (pg *WalletPage) layoutOptionsMenu(gtx layout.Context, optionsMenuIndex int
 
 func (pg *WalletPage) walletSection(gtx layout.Context) layout.Dimensions {
 	return pg.walletsList.Layout(gtx, len(pg.listItems), func(gtx C, i int) D {
+		pg.listLock.Lock()
+		defer pg.listLock.Unlock()
```

Comment:
Copy into a local variable, this defer could have unintended effects.
```suggestion
func (pg *WalletPage) walletSection(gtx layout.Context) layout.Dimensions {
	pg.listLock.Lock()
	listItems := pg.listItems
	pg.listLock.Unlock()
	return pg.walletsList.Layout(gtx, len(listItems), func(gtx C, i int) D {
```